### PR TITLE
add nest thermostat integration

### PR DESCRIPTION
**Pull request by drei01**

### CODE REVIEW

File: api/lib/middleware/nest.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    source: "nest",
+    routes: [
+        {
+            method: "GET",
+            path: "",
+            handler: function(req, res) {
+                var cachedResult = cache.get('nest');
+                if (cachedResult) return res.json(cachedResult);
+				
+				console.log(this.secrets);
```
Comment (danfang) [inline api/lib/middleware/nest.js]: Remove this

File: api/lib/middleware/nest.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+				
+				var url = 'https://developer-api.nest.com/devices/thermostats/' + this.secrets.device_id + '?auth=' + this.secrets.access_token
```
Comment (danfang) [inline api/lib/middleware/nest.js]: Would like to see the url in a constant at the top so it's easier to manipulate if the URL endpoint changes

**Comment by drei01** [comment]
Can be seen in action at http://api.matt-reid.co.uk/nest

**Comment by danfang** [comment]
Awesome, that's great to see. I'm going to try to spend more time cleaning this project up, getting it building and tested.

**Pull request merged**